src/core/server/http/integration_tests/request.test.ts

```diff
@@ -163,24 +163,24 @@ describe('KibanaRequest', () => {
 
   describe('events', () => {
     describe('aborted$', () => {
-      it('emits once and completes when request aborted', async (done) => {
+      it('emits once and completes when request aborted', async () => {
         expect.assertions(1);
         const { server: innerServer, createRouter } = await server.setup(setupDeps);
         const router = createRouter('/');
 
         const nextSpy = jest.fn();
-        router.get({ path: '/', validate: false }, async (context, request, res) => {
-          request.events.aborted$.subscribe({
-            next: nextSpy,
-            complete: () => {
-              expect(nextSpy).toHaveBeenCalledTimes(1);
-              done();
-            },
-          });
 
-          // prevents the server to respond
-          await delay(30000);
-          return res.ok({ body: 'ok' });
+        const done = new Promise<void>((resolve) => {
```

Comment:
promise + callback style API are not supported simultaneously https://github.com/facebook/jest/issues/10529